src/actions/public/watchBlockNumber.ts

```diff
@@ -11,77 +12,129 @@ export type OnBlockNumberFn = (
   prevBlockNumber: OnBlockNumberParameter | undefined,
 ) => void
 
-export type WatchBlockNumberParameters = {
+export type PollOptions = {
   /** Whether or not to emit the missed block numbers to the callback. */
   emitMissed?: boolean
   /** Whether or not to emit the latest block number to the callback when the subscription opens. */
   emitOnBegin?: boolean
+  /** Polling frequency (in ms). Defaults to Client's pollingInterval config. */
+  pollingInterval?: number
+}
+
+export type WatchBlockNumberParameters<
+  TTransport extends Transport = Transport,
+> = {
   /** The callback to call when a new block number is received. */
   onBlockNumber: OnBlockNumberFn
   /** The callback to call when an error occurred when trying to get for a new block. */
   onError?: (error: Error) => void
-  /** Polling frequency (in ms). Defaults to Client's pollingInterval config. */
-  pollingInterval?: number
-}
+} & (GetTransportConfig<TTransport>['type'] extends 'webSocket'
+  ?
+      | {
+          emitMissed?: never
+          emitOnBegin?: never
+          /** Whether or not the WebSocket Transport should poll the JSON-RPC, rather than using `eth_subscribe`. */
+          poll?: false
+          pollingInterval?: never
+        }
+      | (PollOptions & { poll: true })
+  : PollOptions & { poll?: true })
```

Comment:
Should we remove `poll` as an option in this case?

```suggestion
  : PollOptions & unknown)
```

Or setting it to `undefined`

```suggestion
  : PollOptions & { poll?: undefined })
```